File: Codigo/Frontend/src/app/interfaces/invitation.ts
import { Pharmacy } from "./pharmacy";
import { Role } from "./role";

export interface Invitation {
    id: number;
    pharmacy: Pharmacy;
    userName: string;
    role: Role;
    userCode: string;
    isActive: boolean;
}

export class InvitationRequest {
    pharmacy: string;
    userName: string;
    role: string;
    userCode: string;

    constructor(
        pharmacyName: string,
        userName: string,
        roleName: string,
        userCode: string){
            this.pharmacy = pharmacyName;
            this.userName = userName;
            this.role = roleName;
            this.userCode = userCode;
        }
  }

File: Codigo/Backend/PharmaGo.Domain/Entities/Invitation.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace PharmaGo.Domain.Entities
{
    public class Invitation
    {
        public int Id { get; set; }
        public string UserName { get; set; }
        public string UserCode { get; set; }
        public bool IsActive { get; set; } = true;
        public DateTime Created { get; set; }
        public Pharmacy? Pharmacy { get; set; }
        public Role Role { get; set; }

        public Invitation()
        {
            Created = DateTime.Now;
        }
    }
}
